tests/org.jboss.tools.jst.ui.bot.test/src/org/jboss/tools/jst/ui/bot/test/nodejs/NodeJSLauncherTest.java

```diff
@@ -29,74 +28,102 @@
+    
+    private static int APP_PORT = 3000;
+    private static int DEBUG_PORT = 5858;
+    
+    private NodeJSLaunchListener launchListener = null;
+    
+    @BeforeClass
+    public static void prepare() {
+        /* PE is closed in Debug perspective, open it */
+        new ProjectExplorer().open();
+        importExistingProject(IMPORT_PATH);
+        assertTrue(TEST_APP_NAME + " has not been imported!", new ProjectExplorer().containsProject(TEST_APP_NAME));
+        npmInstall(TEST_APP_NAME);
+    }
+    
+    @AfterClass
+    public static void cleanup() {
+        ShellHandler.getInstance().closeAllNonWorbenchShells();
+        new ProjectExplorer().deleteAllProjects();
+    }
+    
+    @Test
+    public void testNodeJSRunAsLauncherAvailable() {
+        ExplorerItem indexJS = new ProjectExplorer().getProject(TEST_APP_NAME).getProjectItem("index.js");
+        indexJS.select();
+        assertTrue("'Run As -> Node.js Application' not available!", runAsNodeJSAppMenu().isEnabled());
+    }
+    
+    @Test
+    public void testNodeJSAppIsRunning() {
+        assertTrue(portAvailable(APP_PORT));
+    	
+        launchListener = new NodeJSLaunchListener();
+        ILaunchManager manager = DebugPlugin.getDefault().getLaunchManager();
+        manager.addLaunchListener(launchListener);
+        
+        ExplorerItem indexJS = new ProjectExplorer().getProject(TEST_APP_NAME).getProjectItem("index.js");
+        indexJS.select();
+        runAsNodeJSAppMenu().select();
+        
+        ConsoleView console = new ConsoleView();
+        console.activate();
+        try {
+            new WaitUntil(new ConsoleHasText("Listening on port 3000"));
+        } catch (WaitTimeoutExpiredException e) {
+            Assert.fail("Node.js App is not running!");
+        }
+        terminate();
+    }
+    
+    @Test
+    public void testNodeJSDebugAsLauncherAvailable() {
+        assertTrue("'Debug As -> Node.js Application' not available!",
+                   debugAsNodeJSAppMenu(new ProjectExplorer().getProject(TEST_APP_NAME).getProjectItem("index.js"))
+                   .isEnabled());
+    }
+    
+    @Test
+    public void testNodeJSAppIsDebugging() {
+    	assertTrue(portAvailable(APP_PORT));
+    	assertTrue(portAvailable(DEBUG_PORT));
+    	
+        launchListener = new NodeJSLaunchListener();
+        ILaunchManager manager = DebugPlugin.getDefault().getLaunchManager();
+        manager.addLaunchListener(launchListener);
+        
+        debugAsNodeJSAppMenu(new ProjectExplorer().getProject(TEST_APP_NAME).getProjectItem("index.js")).select();
+        
+        ConsoleView console = new ConsoleView();
+        console.activate();
+        new WaitUntil(new ConsoleHasText("Debugger listening"), TimePeriod.LONG);
+        
+        assertTrue("Node.js App is not debugging!", console.getConsoleText().contains("Debugger listening"));
+        
+        terminate();
+    }
+    
+    private void terminate(){
```

Comment:
please consider making this "@after" method. That way even if some test fails you can be sure that all appropriate processes were terminated.
If you put @after here then dont forget to make it more robus (ie getNodeJSLaunch can be null, or launchListener can be null too etc)